Week3/vehicle.js

```diff
@@ -0,0 +1,18 @@
+// Create a function called vehicle, like before, but takes another parameter called age, so that vehicle("blue", 1, 5) prints 'a blue used car'
+
+function vehicleType(color,code, age){
```

Comment:
This is also correct, but it's not a good practice to modify the arguments that are passed in the function. Because if you modify a property of that argument in the function,  it will also change the property for that argument outside the function. I know this is not the case in this specific function, but it could happen in the future, which could lead to confusing results. For example:

```
function vehicleType(color,code, anObject){
    if (anObject.age < 1) {
        anObject.age = 'bused'
    } else {
       anObject.age = ' new'
    }

    if (code == 1){
        console.log('a ' + color + anObject.age + ' car');
    } else if (code == 2) {
        console.log('a ' + color + anObject.age + ' motorbike');
    }
}

let color = "green";
let code = 1;
let anObject = { age: 12, name: "test" };

console.log("before the function, the age is " + anObject.age);
vehicleType(color, code, anObject);
console.log("after the first function, the age is " + anObject.age);
```

In this code example, the property `age` is modified in the function, which affects the `anObject` variable, which is outside the function.